# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: 9213b6bda67a
# Full Hash: 9213b6bda67a689644f442f7b98902c5a5184c74
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-09-24 09:46:13
# Regressor Bug: 1725572
# File Overlap Count: 1
# Description:
#   Bug 1725572 - Part 2: Keep BrowsingContext alive until every process has acknowledged the discard, r=farre
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D125897
# ==============================================================================

diff -r 6bb38c7309e5 -r 9213b6bda67a dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Thu Sep 23 18:54:31 2021 +0000
+++ b/dom/ipc/PContent.ipdl	Thu Sep 23 18:54:31 2021 +0000
@@ -1790,13 +1790,15 @@
     async CreateBrowsingContext(uint64_t aGroupId, BrowsingContextInitializer aInit);
 
     /**
-     * Discards the passed-in BrowsingContext. If the BrowsingContext has
-     * already been discarded, this message does nothing.
-     * The response promise is fulfilled when the process has flagged the
-     * BrowsingContext as discarded.
+     * If aDoDiscard is true, discards the passed-in BrowsingContext. If the
+     * BrowsingContext has already been discarded, this message does nothing.
+     * If the receiver is the parent process, resolves when all content
+     * processes have flagged the BrowsingContext as discarded, and if the
+     * receiver is a child process, resolves when that child process has flagged
+     * the BrowsingContext as discarded.
      */
-    async DiscardBrowsingContext(MaybeDiscardedBrowsingContext aContext)
-            returns (bool unused);
+    async DiscardBrowsingContext(MaybeDiscardedBrowsingContext aContext, bool aDoDiscard)
+            returns (uint64_t unused);
 
     async AdjustWindowFocus(MaybeDiscardedBrowsingContext aContext,
                             bool aIsVisible, uint64_t aActionId);
